src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensRangeHandler.cs

```diff
@@ -50,6 +50,13 @@ public SemanticTokensRangeHandler(IGlobalOptionService globalOptions)
 
             var options = _globalOptions.GetClassificationOptions(context.Document.Project.Language);
 
+            // If the project's compilation isn't yet available, kick off a task in the background to
+            // hopefully make it available faster since we'll need it later to compute tokens.
+            if (!context.Document.Project.TryGetCompilation(out _))
+            {
+                _ = context.Document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
This is supposed to be handled by `BackgroundCompiler` and `CompilationAvailableTaggerEventSource`. Is there a reason we want it here separately?

Reply:
There was an issue in Razor where `document.Project.TryGetCompilation(out var _)` was returning false even after ~5 mins of waiting. We thought kicking off a separate task at the beginning might solve this problem and allow compilation to return sooner.

cc @CyrusNajmabadi is this the correct way to handle things?